src/csvReaderWrapper.js

```diff
@@ -19,23 +19,35 @@ export const parseCSV = (csvfile, download, useDefaultColumnDataType) => {
   });
 };
 
-const isCellValid = (cell) => {
-  return cell !== undefined && cell !== "";
-}
+const cleanData = (data) => {
+  var cleanedData = []
 
-const isRowValid = (row) => {
-  var cells = Object.values(row);
-  return cells.every(isCellValid)
-}
+  for (var row of data) {
+    var cleanedRow = tryGetCleanedRow(row);
+    if (cleanedRow !== null) {
+      cleanedData.push(cleanedRow);
+    }
+  }
 
-const cleanData = (data) => {
-  var cleanedData = data.filter(row => isRowValid(row));
   return cleanedData;
 }
 
-const countRemovedRows = (originalData, cleanedData) => {
-  var removedRowsCount = originalData.length - cleanedData.length;
-  store.dispatch(setRemovedRowsCount(removedRowsCount));
+const tryGetCleanedRow = (row) => {
+  for (var cellName in row) {
+    if (!isCellValid(row[cellName])) {
+      return null;
+    }
+
+    if (typeof row[cellName] === 'string') {
```

Comment:
Pardon my ignorance, but are numerical cells actually numbers by this point?

Reply:
At this point, all values are converted to strings. I added this in because I accidentally uploaded an .xlsx file and got an error related to the type for `trim()`. I also thought it'd be good practice to program defensively since `trim()` can only be performed on a string. What do you think?

As a side note, I noticed .xls and .xlsx are supported file types for the file upload button in `SelectDataset`, but .xlsx files don't display the data correctly. Do we ever plan on supporting these files? The file upload button specifically mentions CSV, but just wondering if we should prevent accidental uploads of .xlsx files for users (like me).

Reply:
It might be worth a comment explaining that this shouldn't ever be a string, otherwise it struck me that we might be expecting that to sometimes be the case here.

You make a really good point about non-CSV files, and something we should perhaps deal with a bit more proactively than we have.  It's out of scope for this PR, but should we try to catch non-CSV file issues earlier, rather than just deal with one of the symptoms in this cleanup function?  

And great catch about .xls & .xlsx files.  The former appear to be binary, and the latter a zip full of xml files.  We support neither, so we should remove them both from the upload accept filter.

Reply:
Looking at the Papa parse documentation, it looks like it always parses the values in to a string. @Erin007 can correct me if I'm wrong. I'll remove the string check since this seems redundant.

Reply:
Also, thank you for looking in to the non-CSV file issue! I made a Trello card for it [here](https://trello.com/c/Pqn0zB6M/260-prevent-non-csv-file-uploads). I think it's a very small change, but I'll put it in its own PR to keep things organized.